### PR TITLE
[5.8] Bugfix - QueueFake size method and added a test

### PR DESCRIPTION
**Pull request by hlorofos**
This is fixing https://github.com/laravel/framework/issues/29755

### CODE REVIEW

File: tests/Queue/QueueSizeTest.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+use Illuminate\Bus\Queueable;
+use Orchestra\Testbench\TestCase;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class QueueSizeTest extends TestCase
+{
+    public function test_queue_size()
+    {
+        \Queue::fake();
```
Comment (GrahamCampbell) [inline tests/Queue/QueueSizeTest.php]: This code isn's going to work. There's no such class.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Queue/QueueSizeTest.php (hunk 1 above)
Comment (driesvints) [reply tests/Queue/QueueSizeTest.php]: You'll need to reference the facade class.

Comment (hlorofos) [reply tests/Queue/QueueSizeTest.php]: @GrahamCampbell, then how on earth this is working locally and on travisCI?
```
vendor/bin/phpunit tests/Queue/QueueSizeTest.php 
PHPUnit 7.5.15 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.1.25 with Xdebug 2.5.3
Configuration: /Users/j/Documents/Sponge/snaptest/2/framework/phpunit.xml.dist

.                                                                   1 / 1 (100%)

Time: 146 ms, Memory: 12.00 MB

OK (1 test, 4 assertions)
```

Comment (GrahamCampbell) [reply tests/Queue/QueueSizeTest.php]: Probably because the class alias is setup.

File: tests/Queue/QueueSizeTest.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Illuminate\Tests\Integration\Queue;
+
+use Queue;
```
Comment (driesvints) [inline tests/Queue/QueueSizeTest.php]: You need to reference the actual facade class:
```suggestion
use Illuminate\Support\Facades\Queue;
```

File: tests/Queue/QueueSizeTest.php [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 namespace Illuminate\Tests\Integration\Queue;
 
+use Queue;
```
Comment (GrahamCampbell) [inline tests/Queue/QueueSizeTest.php]: This needs to be the actual name of the facade please. `Illuminate\Support\Facades\Queue`.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Queue/QueueSizeTest.php (hunk 3 above)
Comment (hlorofos) [reply tests/Queue/QueueSizeTest.php]: okay, I just was inspired by /tests/Integration/Queue/QueuedListenersTest.php ;-)

Comment (driesvints) [reply tests/Queue/QueueSizeTest.php]: @hlorofos that's an integration test. The facades are available there. This is a unit test so the framework isn't loaded.

Comment (hlorofos) [reply tests/Queue/QueueSizeTest.php]: @driesvints thanks for your explanations!

**Comment by GrahamCampbell** [comment]
This will break if the queue is null?

**Comment by hlorofos** [comment]
> This will break if the queue is null?

@GrahamCampbell thats why I put first assertion in the test without pushing anything to a queue:

```
Queue::fake();
        $this->assertEquals(0, Queue::size());
```

there is initializer for `$job` in `QueueFake`
```
    protected $jobs = [];
```

**Comment by GrahamCampbell** [comment]
No, I mean if there are jobs, but the queue name is null, it should be non-zero.

**Comment by GrahamCampbell** [comment]
To clarify, in Laravel, using null for the queue name means "use the default". This needs to be handled within the implementation.

**Comment by hlorofos** [comment]
> To clarify, in Laravel, using null for the queue name means "use the default". This needs to be handled within the implementation.

@GrahamCampbell , here are my explanations in the test, please consider if possible to explain what's wrong based on the test:

```
        dispatch($job); // Pushed to default queue
        dispatch(new TestJob2); //  Other job pushed to default queue, this would validate we traverse through different job classes properly
        dispatch($job)->onQueue('Q2'); // Pushed to queue 'Q2'

        $this->assertEquals(2, Queue::size()); // Get the amount of jobs in default queue, internally its NULL for QueueFake
        $this->assertEquals(1, Queue::size('Q2')); // Get the amount of jobs in queue `Q2`
``` 

If you're talking about `default` property for Queue implementations this have nothing to do with QueueFake implementation.

**Comment by hlorofos** [comment]
@taylorotwell you interpret this wrong, I'm testing my code which expose Queue::size through API. Since I'm faking queue and testing API which calling size() - as a developer I expect `size` method  return proper value instead of some random number.

**Comment by driesvints** [comment]
@hlorofos we got this merged 👍

**Comment by hlorofos** [comment]
> @hlorofos we got this merged 👍

@driesvints appreciate all your help along the way!

**Pull request merged**